apps/nightscout/nightscout.star

```diff
@@ -2,154 +2,578 @@
 Applet: Nightscout
 Summary: Shows Nightscout CGM Data
 Description: Displays Continuous Glucose Monitoring (CGM) data from the Nightscout Open Source project (https://nightscout.github.io/).
-Author: Jeremy Tavener
+Authors: Jeremy Tavener, Paul Murphy
 """
 
 load("render.star", "render")
 load("http.star", "http")
 load("time.star", "time")
 load("encoding/base64.star", "base64")
 load("encoding/json.star", "json")
+load("encoding/csv.star", "csv")
 load("cache.star", "cache")
 load("schema.star", "schema")
 load("re.star", "re")
+load("humanize.star", "humanize")
+load("sunrise.star", "sunrise")
 
-COLOR_RED = "#f00"
-COLOR_YELLOW = "#ff0"
-COLOR_GREEN = "#0f0"
-COLOR_GREY = "#666"
+COLOR_RED = "#C00"
+COLOR_DARK_RED = "#911"
+COLOR_YELLOW = "#ff8"
+COLOR_ORANGE = "#d61"
+COLOR_GREEN = "#2b3"
+COLOR_GREY = "#777"
 COLOR_WHITE = "#fff"
+COLOR_NIGHT = "#444"
 
-DEFAULT_NORMAL_HIGH = 150
+DEFAULT_NORMAL_HIGH = 180
 DEFAULT_NORMAL_LOW = 100
 DEFAULT_URGENT_HIGH = 200
 DEFAULT_URGENT_LOW = 70
 
-CACHE_TTL_SECONDS = 300
+DEFAULT_SHOW_GRAPH = "true"
+DEFAULT_SHOW_CLOCK = "true"
+DEFAULT_NIGHT_MODE = "false"
+GRAPH_WIDTH = 43
+GRAPH_BOTTOM = 50
+GRAPH_TOP = 275
+
+CACHE_TTL_SECONDS = 1800  #30 mins
```

Comment:
Can you help me better understand this data? Is 30 min cached acceptable? I'm going to bump this back to 5 minutes just to play it safe. Basically, this is our most sensitive app and I don't know the full ramifications of someone getting 30 minute stale data from Nightscout and what the implications are. Feel free to follow up with an additional PR to bump it up with reasoning on why it's ok to be stale.
```suggestion
CACHE_TTL_SECONDS = 300  #5 mins
```

Reply:
Actually, we want this to be 30 minutes. The way it works is that the cache is used only if live data is not available. So if Nightscout is unavailable or there's not a recent reading, the app will show the last reading for up to but not more than 30 minutes.